src/QsCompiler/Tests.Compiler/TestCases/GlobalVerification.qs

```diff
@@ -365,6 +364,20 @@ namespace Microsoft.Quantum.Testing.GlobalVerification {
         }
     }
 
+    operation AllPathsReturnValue13 () : Int {
+        within {
+            return 1;
+        }
+        apply {}
+    }
+
+    operation AllPathsReturnValue14 () : Int {
+        within {}
+        apply {
+            return 1;
```

Comment:
Similarly here, what are the semantics for when `within` isn't empty?

Reply:
This is valid code currently, since it seemed that the behavior seems fairly clear (it actually returns and does not execute further - same as if you return somehere else in a statement). But let's have a quick chat on whether that makes sense to keep, or if there are any reasons why we would want to give an error in this case.

Reply:
After some thinking: I think the right thing to do here is to limit the use of return statements to be at the end of an execution path in the apply block - same as for returns from within using and borrowing - and guarantee that the adjointed within block is executed before returning (treat is as a try/finally, same as deallocating qubits). However, to limit the burden on a runtime, I would like to not support it until we have the proper translation. I will thus disallow it for now.